src/components/MessageInput/hooks/utils.ts

```diff
@@ -186,10 +188,29 @@ export const checkUploadPermissions = async <
     );
 
     if (blocked) {
-      sendErrorNotification();
+      sendNotAllowedErrorNotification();
       return false;
     }
   }
 
+  const sizeLimit = size_limit || DEFAULT_UPLOAD_SIZE_LIMIT_BYTES;
+  if (file.size && file.size > sizeLimit) {
+    addNotification(
+      t('File is too large: {{ size }}, maximum upload size is {{ limit }}', {
+        limit: prettifyFileSize(sizeLimit),
+        size: prettifyFileSize(file.size),
+      }),
+      'error',
+    );
+    return false;
+  }
+
   return true;
 };
+
+function prettifyFileSize(bytes: number) {
```

Comment:
We  could reuse this function also for file size in message attachments.

Reply:
Oh no, my mistake - we do. Will update.

Reply:
Done in https://github.com/GetStream/stream-chat-react/pull/2301/commits/5aba95cad36b42cf8630020ec91a89668624eaf1